File: CodeTester-IntelliJ-IDEA/src/main/java/de/fxnm/action/BaseAction.java
package de.fxnm.action;

import com.intellij.openapi.actionSystem.AnAction;
import com.intellij.openapi.actionSystem.AnActionEvent;
import com.intellij.openapi.actionSystem.Presentation;
import com.intellij.openapi.diagnostic.Logger;
import com.intellij.openapi.project.Project;

import org.jetbrains.annotations.NotNull;

import java.util.Optional;

import static java.util.Optional.ofNullable;

public abstract class BaseAction extends AnAction {

    private static final Logger LOG = Logger.getInstance(BaseAction.class);

    @Override
    public void update(final @NotNull AnActionEvent event) {
        try {
            final Project project = getEventProject(event);
            final Presentation presentation = event.getPresentation();

            if (project == null) {
                presentation.setEnabled(false);
                presentation.setVisible(false);
                LOG.warn("Action update failed but keep secret");
                return;
            }

            super.update(event);
            presentation.setVisible(true);

        } catch (final Throwable e) {
            LOG.warn("Action update failed", e);
        }
    }

    protected Optional<Project> project(@NotNull final AnActionEvent event) {
        return ofNullable(getEventProject(event));
    }
}

File: CodeTester-IntelliJ-IDEA/src/main/kotlin/de/fxnm/action/BaseAction.kt
package de.fxnm.action

import com.intellij.openapi.actionSystem.AnAction
import com.intellij.openapi.actionSystem.AnActionEvent
import com.intellij.openapi.diagnostic.Logger
import com.intellij.openapi.project.Project
import de.fxnm.util.CodeTesterBundle
import java.util.*

abstract class BaseAction : AnAction() {

    override fun update(event: AnActionEvent) {
        try {
            val project = getEventProject(event)
            val presentation = event.presentation

            if (project == null) {
                presentation.isEnabled = false
                presentation.isVisible = false
                LOG.warn(CodeTesterBundle.message("plugin.action.base.updateFailed.silent"))
                return
            }

            super.update(event)
            presentation.isVisible = true
        } catch (e: Throwable) {
            LOG.warn(CodeTesterBundle.message("plugin.action.base.updateFailed"), e)
        }
    }

    protected fun project(event: AnActionEvent): Optional<Project> {
        return Optional.ofNullable(getEventProject(event))
    }

    companion object {
        private val LOG = Logger.getInstance(BaseAction::class.java)
    }
}
